### DISCUSSION TITLE
Function to check if a sequence is a valid degree sequence

### DISCUSSION THREAD

**Discussion by vaswin0** [networkit/networkit#1239] [Q&A]
I am not aware of a function that will check if a sequence is a valid degree sequence. It would be nice to have one such.

**Comment by fabratu** [comment]
Hey,

likely you are referring to degree sequence-based graph generators? In that case, both Havel Hakimi[1] and Edge Switching[2] support the function `isRealizable()`. For the function to be used, you need to create an object for the generator and then call the function.

Let me know, whether this suits your needs here.

```
import networkit as nk
hhg = nk.generators.HavelHakimiGenerator([5,4,...])
hhg.isRealizable()
```

[1] https://networkit.github.io/dev-docs/python_api/generators.html#networkit.generators.HavelHakimiGenerator.isRealizable
[2] https://networkit.github.io/dev-docs/python_api/generators.html#networkit.generators.EdgeSwitchingMarkovChainGenerator.isRealizable

**Comment by vaswin0** [comment]
This seems like it. Thanks
